packages/fiori-annotation-api/test/unit/cds/writer.test.ts

```diff
@@ -772,6 +772,115 @@ annotate S.E with {
+        {
+            $Type : 'UI.DataField',
+            Value : test,
+            Label : 'testLabel',
+        },
+    ];`
+                );
+            });
+            test('by default inserting at the end when deleting property - datapoint', async () => {
+                const fixture = `
+    Service S { entity E {}; };
+    annotate S.E with @UI.DataPoint  : {
+        $Type : 'UI.DataPointType',
+        Value : ID,
+        Criticality: #Critical
+    };`;
+                await testWriter(
+                    fixture,
+                    [
+                        {
+                            type: 'insert-record-property',
+                            pointer: '/targets/0/assignments/0/value',
+                            element: createElementNode({
+                                name: Edm.PropertyValue,
+                                attributes: {
+                                    [Edm.Property]: createAttributeNode(Edm.Property, 'CriticalityLabels'),
+                                    [Edm.String]: createAttributeNode(Edm.String, 'testLabel')
+                                }
+                            })
+                        },
+                        {
+                            type: 'delete-record-property',
+                            pointer: '/targets/0/assignments/0/value/properties/2'
+                        }
+                    ],
+                    `
+    Service S { entity E {}; };
+    annotate S.E with @UI.DataPoint  : {
+        $Type : 'UI.DataPointType',
+        Value : ID,
+        CriticalityLabels : 'testLabel',
+    };`
+                );
+            });
+            test('by default inserting at the end when deleting only property - datapoint2', async () => {
+                const fixture = `
+    Service S { entity E {}; };
+    annotate S.E with @UI.DataPoint  : {
+        $Type : 'UI.DataPointType'
+    };`;
+                await testWriter(
+                    fixture,
+                    [
+                        {
+                            type: 'insert-record-property',
+                            pointer: '/targets/0/assignments/0/value',
+                            element: createElementNode({
+                                name: Edm.PropertyValue,
+                                attributes: {
+                                    [Edm.Property]: createAttributeNode(Edm.Property, 'CriticalityLabels'),
+                                    [Edm.String]: createAttributeNode(Edm.String, 'testLabel')
+                                }
+                            })
+                        },
+                        {
+                            type: 'delete-record-property',
+                            pointer: '/targets/0/assignments/0/value/properties/0'
+                        }
+                    ],
+                    `
+    Service S { entity E {}; };
+    annotate S.E with @UI.DataPoint  : {
+        
```

Comment:
It looks like we have some extra whitespace here. Is it difficult to remove it?

Reply:
Its not an easy fix if i call the parent range to delete can cause overlapping edits. Will address in another PR